### PR TITLE
Added 'Does Not Tip' backdoor traitor item

### LINKED ISSUE #18402: Feature Request: "Doesn't Tip Note" Traitor item
-Doesn't Tip Note
\>Chef / Quartermaster Traitor Item
\>Costs 10 TC's
\>Upon Purchasing it, all cargo pizza orders for the rest of the round are sabotaged

### PR DESCRIPTION
**Pull request by DamianX**
Closes #18402

Disguised as a PDA, lets you put the station on or off the "Does Not Tip" list.
Being on the list means all pizza orders will be poisoned in various ways.

- All nutriment is removed from the pizza
- Pizza can contain virii, toxic chemicals, mutagen, or strong acids
- Costs 10 TC

Extras: 
- Added a `unique` var for uplink item datums, if set only one item of this kind can be bought during the round. This could be a lame idea.

:cl:
 * rscadd: Added a new traitor item: "Does Not Tip" database backdoor. Once purchased, lets you add the station to Central Command's "Does Not Tip" database. Stations on the list get extra condiment with their pizza.

### CODE REVIEW

**Comment by Irradiation** [comment]
Jokes on you, I've perfected my cargo strats to order pizza in 15 seconds. YOU CAN'T CATCH MEEEEE

**Comment by Durgen** [comment]
10 tc seems a bit too expensive

**Comment by SonixApache** [comment]
if you consider the amount of pizza that gets ordered during a normal round, i'd say it's worth it

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -2952,6 +2955,22 @@
...
+	var/static/list/possible_poisons = list(
+		BLEACH,
+		PLASMA,
+		TOXIN,
+		SOLANINE,
+		PLASTICIDE,
+		RADIUM,
+		CRYPTOBIOLIN,
+		IMPEDREZENE,
+		SOYSAUCE
```
Comment (Durgen) [inline code/modules/reagents/reagent_containers/food/snacks.dm:2970]: how about XENOMICROBES?

**Comment by Kurfursten** [comment]
Just putting this out there: let anyone buy this, not just chefs. Too many of the job specific tatortot items are just a matter of "oh the pizza's poisoned, lynch the chef". Since only one person can buy this tatortot item anyway, let anyone poison pizza orders. As Dan puts it, "there's nothing better than a bite of mystery".

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (Irradiation) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: HOW ABOUT AUTISM NANITES, HUH?

**Comment by DamianX** [comment]
I'm not opposed to the idea at all but that could be said about any job-specific item, why this one?

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (DamianX) [reply code/modules/reagents/reagent_containers/food/snacks.dm:2970]: go away

**Comment by Kurfursten** [comment]
A lot of job specific traitor items are very active-participation. For example with the power gloves, syndie wheelchair, false-bottom briefcase with SMG, banana esword, mime gun, or radgun, you are right there in the action murdering people. So if you don't finish off your targets personally, you're outed anyway.

This is more like a mystery item like the singulo beacon, power sink, or a trail of emagged doors. You find it after the symptoms set in and try to investigate to figure out who is responsible. For example, maybe if you hunt down and destroy the "does not tip" PDA it will restore the station's pizza orders. You can make this same argument for some other items like the syndicate sauce (chef), bycarodyne kit (chemist), pickpocket gloves (assistant), contortionist's jumpsuit (atmos tech), etc. but I wouldn't be opposed to reappraising their availability (in a separate PR/s) as well.

**Comment by Jubisloviu** [comment]
BASED

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -2973,6 +2992,24 @@
...
+		"donor" = null,
+		"viruses" = null,
+		"blood_DNA" = null,
+		"blood_type" = "AB+",
+		"resistances" = null,
+		"trace_chem" = null,
+		"virus2" = list()
+	)
+	blood_data["virus2"]["[new_virus.uniqueID]"] = new_virus
+	reagents.add_reagent(BLOOD, 56, blood_data)
```
Comment (jknpj) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: 56?

Comment (DamianX) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: What might appear to be a magic number is in fact the amount of reagents in the original pizza. I'll make it clearer.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 3]
```diff
@@ -3034,6 +3079,10 @@
 	reagents.add_reagent(IMIDAZOLINE, 12)
 	bitesize = 2
 
+/obj/item/weapon/reagent_containers/food/snacks/sliceable/pizza/vegetablepizza/make_poisonous()
+	reagents.clear_reagents()
+	reagents.add_reagent(MUTAGEN, 48)
```
Comment (jknpj) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: 48?

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 2 above)
Comment (jknpj) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: I see... *define when*?

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (Killette2) [reply code/modules/reagents/reagent_containers/food/snacks.dm:2970]: no mindbreaker?

Comment (DamianX) [reply code/modules/reagents/reagent_containers/food/snacks.dm:2970]: it's in the mushroom pizza

File: code/datums/uplink_item.dm [hunk 4]
```diff
@@ -56,6 +56,8 @@ var/list/uplink_items = list()
 	var/list/job = null
 	var/only_on_month	//two-digit month as string
 	var/only_on_day		//two-digit day as string
+	var/unique = FALSE	// Can only be bought once, globally
+	var/static/times_bought = 0
```
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [inline code/datums/uplink_item.dm]: lmao

**Pull request merged**